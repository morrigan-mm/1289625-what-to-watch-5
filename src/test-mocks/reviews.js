const reviews = [
  {
    id: 1,
    text: `some review text 1`,
    rating: 1.1,
    author: `Author name1`,
    reviewDate: `2020-01-01T14:13:56.569Z`
  },
  {
    id: 2,
    text: `some review text 2`,
    rating: 2.2,
    author: `Author name2`,
    reviewDate: `2020-02-02T14:13:56.569Z`
  },
  {
    id: 3,
    text: `some review text 3`,
    rating: 3.3,
    author: `Author name3`,
    reviewDate: `2020-03-03T14:13:56.569Z`
  }
];

export default reviews;
