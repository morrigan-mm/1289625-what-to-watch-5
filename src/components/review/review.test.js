import React from "react";
import renderer from "react-test-renderer";
import reviews from "../../test-mocks/reviews";
import Review from "./review";

const [reviewMock] = reviews;

describe(`Review`, () => {
  it(`Should render correctly`, () => {
    const tree = renderer
      .create(
          <Review review={reviewMock} />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
