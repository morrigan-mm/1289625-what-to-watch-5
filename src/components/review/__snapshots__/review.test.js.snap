// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Review Should render correctly 1`] = `
<div
  className="review"
>
  <blockquote
    className="review__quote"
  >
    <p
      className="review__text"
    >
      some review text 1
    </p>
    <footer
      className="review__details"
    >
      <cite
        className="review__author"
      >
        Author name1
      </cite>
      <time
        className="review__date"
        dateTime="1/1/2020"
      >
        January 1, 2020
      </time>
    </footer>
  </blockquote>
  <div
    className="review__rating"
  >
    1.1
  </div>
</div>
`;
